### ISSUE TITLE
Segfault on query when constraint used within subquery (<= 1.6.0)

### ISSUE THREAD

**Issue by 0xdabbad00** [facebook/osquery#1715]
I've not looked into this deeper other than trying to minimize my query.  If you run the following query from osqueryi you'll get a `Segmentation fault: 11`:
```
select * from (select total_seconds from uptime where total_seconds > 1) 
union all select total_seconds from uptime;   
```
If you remove the `where` clause, it successfully returns something like:
```
+---------------+
| total_seconds |
+---------------+
| 177760        |
| 177760        |
+---------------+
```

This was run on OSX 10.11, osquery 1.6.0.

**Comment by theopolis** [comment]
Hey @0xdabbad00, can you try using: `1.6.1-34`? We bumped the version of SQLite from 3.8 to 3.10. I am running that version and the subquery with predicate works fine.

I'm bisecting now to find when it was fixed.

**Comment by theopolis** [comment]
Actually, this is fixed in `1.6.1`, now trying to bisect between `1.6.1

**Comment by theopolis** [comment]
Survey (bisect) says the fix was: #1675
